src/Brick/Widgets/List.hs

```diff
@@ -545,18 +545,34 @@ splitClamp l i =
         -- is in the list, so we don't need to know the length
         clamp 0 (if null t then length (l ^. listElementsL) - 1 else i) i
 
--- | Set the selected index for a list to the index of the specified
--- element if it is in the list, or leave the list unmodified otherwise.
+-- | Set the selected index for a list to the index of the first
+-- occurence of the specified element if it is in the list, or leave
+-- the list unmodified otherwise.
 --
--- Complexity: same as 'traverse' for the container type (typically
--- /O(n)/).
-listMoveToElement :: (Eq e, Traversable t)
+-- /O(n)/.  Only evaluates as much of the container as needed.
+listMoveToElement :: (Eq e, Foldable t, Splittable t)
                   => e
                   -> GenericList n t e
                   -> GenericList n t e
-listMoveToElement e l =
-    let i = fmap fst $ find ((== e) . snd) $ imap (,) (l^.listElementsL)
-    in l & listSelectedL %~ (i <|>)
+listMoveToElement e = listSeekToElement (== e) . set listSelectedL Nothing
+
+-- | Set the selected index to the next element matching the
+-- predicate.  If there is no selected element, the search starts at
+-- the beginning.  If no matching element is found, leave the list
+-- unmodified.
+--
+-- /O(n)/.  Only evaluates as much of the container as needed.
+listSeekToElement :: (Foldable t, Splittable t)
+                  => (e -> Bool)
+                  -> GenericList n t e
+                  -> GenericList n t e
+listSeekToElement test l =
+  let
```

Comment:
Please use four-space indents and write `let` bindings as written elsewhere in this module.